### ISSUE TITLE
"Move to view" no longer working when moving to another dashboard after updating to 2024.3

### ISSUE THREAD

**Issue by erikgeurts** [home-assistant/frontend#20460]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

After updating to 2024.3, the "move to view" action to move a card to another view that's part of another dashboard no longer works. The dialog to select the dashboard and the view I want as destination shows up fine, but when I click "move", it simply closes and nothing happens. Moving a card to a view of the same dashboard still works.

### Describe the behavior you expected

I expect to be able to move cards to whatever view under whatever dashboard.

### Steps to reproduce the issue

1. Click the "move to view" command for a card while in edit mode
2. Select a view of another dashboard than the one you're currently editing
3. Click "move"

### What version of Home Assistant Core has the issue?

core-2024.3.3

### What was the last working version of Home Assistant Core?

core-2024.2.3

### In which browser are you experiencing the issue with?

Chrome OS v123

### Which operating system are you using to run this browser?

Chrome OS

### State of relevant entities

```txt
n/a
```

### Javascript errors shown in your browser console/inspector

```txt
no errors found
```

**Comment by ildar170975** [comment]
> but when I click "move", it simply closes and nothing happen

In my case something happened - the full dashboard became a copy of another dashboard.
https://github.com/home-assistant/frontend/issues/20462

**Comment by karwosts** [comment]
Please note the issue template requires you to reproduce on the latest version, which is 2024.4.

When I tried this on 2024.4, it seems to work fine for me. Please try again after updating and confirm.

**Comment by erikgeurts** [comment]
I'll test this as soon as I've had a chance to update to v2024.4, but I'll wait a little while more until the bug fix releases have stopped coming in.

**Comment by erikgeurts** [comment]
After updating to 2024.4, the 'move to view' feature works correctly again, also when moving a card from a subview in one dashboard to a subview in another dashboard. I'm assuming something was temporarily broken due to the introduction of the new sections functionality recently, and subsequently fixed.

**Comment by ildar170975** [comment]
Suggest to reopen the issue.
I still cannot move cards from one dashboard to another.
2024.4.3

**Comment by erikgeurts** [comment]
I'm at 2024.4.4 now, and it works just fine for me. I'm happy to re-open this issue, but as far as I can tell it has been resolved.

**Comment by ildar170975** [comment]
> I can tell it has been resolved

Means: you do not observe it any more.
Thanks for reopening.

**Comment by ChristophCaina** [comment]
question:

Was the dashboard you've selected to move your content to one of the new sections type?
I've noticed something similar after moving a card from a "normal / old" dashboard to a new Sections dashboard.

**Comment by erikgeurts** [comment]
Not, it happened when trying to move a card from a subview to another subview under a different dashboard, both the origin and the target were traditional types.

**Comment by ildar170975** [comment]
In my case there were normal views (not subviews).

**Comment by erikgeurts** [comment]
This issue has resurfaced for me, I am now once again unable to move a card from a subview in one dashboard to a subview in another dashboard.

**Comment by charlespick** [comment]
I am getting this issue when moving items from another dashboard to "overview" or "default" but I can move cards from the default view back. This gets posted to the browser console every time I try:
```
Uncaught (in promise) TypeError: right-hand side of 'in' should be an object, got undefined
    a https://hass.makerland.xyz/frontend_latest/20095.Hx4uiEB8mIE.js:1
    viewSelectedCallback https://hass.makerland.xyz/frontend_latest/82635.lBavD54VUqo.js:1
    value https://hass.makerland.xyz/frontend_latest/64351.P1RuEn9YmgY.js:5
    handleEvent https://hass.makerland.xyz/frontend_latest/app.o3bCeFDd3rk.js:2
```

**Comment by karwosts** [comment]
Fixed by #21142

This bug only manifested when moving a card to a view on a different dashboard that was at a higher index than the number of views on the current dashboard, that's why it seems to work some times and not others.

To workaround till next release, you can move your card in two steps:
 - move the card to the first view on the desired dashboard
 - then move it a second time to the desired view on the current dashboard
 
Think that should work.

**Comment by erikgeurts** [comment]
I'm happy to be able to report that this is now working correctly again. Thanks to everyone!
